proj3_chess/chess_pieces.py

```diff
@@ -0,0 +1,130 @@
+from board import Board
+from player import Player
+
+
+class ChessPiece:
+    """Mother class for all chess pieces."""
+
+    piece_name = ""
+
+    def __init__(self, player: Player, board: Board, spot: tuple) -> None:
+        # TODO: raise error if spot is already taken
+        self.player = player
+        self.player.pieces.append(self)  # add the piece to the player
```

Comment:
Editing a different class in the constructor of a class is weird. you want your classes to have as little interactions as possible.

Reply:
I'd even argue that a `ChessPiece` does not need to save a reference to the player at all, as it can exist without the concept of a player. You can pass it a `move_direction` and `name` in the constructor

Reply:
I was actually wondering about this. I guess a chesspiece can exist without a player but getting a player instead of separate values seemed like a good way to connect the two and make sure the graphic symbolization and the direction are consistent to all pieces, while also allowing you to have more than 2 players on the same board, specifically ones with the same move direction (I use the player to determine weather or not the move is leagal and if the nive captures a piece). I guess I can find a wierd way to get around this but my hunch is that it would make things even uglier and more unreadable.

Reply:
About the first one: I really didn't know what to do about ` Board ` . My logic kept telling me that a piece without a board is meaningless.
I can add the pieces from within the board class, but that would also be wierd: the pieces still depend on board for checking if the move is legal (though only reading if it help somehow) and you'd still need to add the pieces to the board seperatly just through a different class. 
Also, it makes sense that in order to move a piece you'll call a function within the ` ChessPiece `  and not through the board. That means that all changes to the boards anyway happen through the pieces. Or is it better to move them through the board?